src/repo/reservoirRepo.js

```diff
@@ -1,18 +1,42 @@
 import qs from 'qs'
 import { capitalize } from '@/lib/primitive-helpers'
 
-const formatTimeSeries = (id, timeSeries) => {
-  const valueName = `${timeSeries[0]?.name?.split('_').map(capitalize).join(' ')} (#${id})`
+export default function (axios) {
+  return {
+    // Get reservoir by id (fid)
+    getById: id => axios.$get(`reservoir/${id}`),
+
+    // Get only the service water area (raw data)
+    getTimeSeriesById: id =>
+      axios
+        .$get(`reservoir/${id}/ts/surface_water_area`)
+        .then(timeSeries => formatTimeSeries(id, timeSeries)),
 
-  // TODO: make sure this km2 comes from the backend again as an unit
-  // const valueUnit = timeSeries[0]?.unit
-  const valueUnit = 'km2'
+    getByGeometry: geometry =>
+      axios.post('reservoir/geometry', geometry).then(({ data }) => data),
 
-  const data = timeSeries.map(({ t, value: valueInM2 }) => {
-    const value = (valueInM2 / 1000000).toFixed(2)
-    return [t, value]
-  })
+    getTimeSeriesByGeometry: (geometry, variableName, period) =>
+      axios.post(`reservoir/geometry/ts/${variableName}?agg_period=${period}`, geometry).then(formatMultipleTimeSeries),
+
+    getByIds: ids => axios.$get(`reservoir/?${qs.stringify({ ids, limit: ids.length }, { indices: false })}`),
+
+    getTimeSeriesByIds: (ids, variableName, period) => {
+      return axios
+        .$get(`ts/?${qs.stringify({
+          reservoir_ids: ids,
+          variable_name: variableName,
+          agg_period: period,
+        }, { indices: false })}`)
+        .then(formatMultipleTimeSeries)
+    },
+  }
+}
 
```

Comment:
I could also move all this to a different file, just not sure where

Reply:
We could put it into lib as helpers but we only use them here so it should be fine